2017-11-03  Joel John George  <george@lnxsrv09.seas.ucla.edu>

        * timespec.h (timespec_cmp): Updated comparison in timespec_cmp to
        prevent potential integer overflow from occurring.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2017-11-03 23:05:48.785640000 -0700
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec)
+    int diff = 1;
+  else if(a.tv_sec < b.tv_sec)
+    int diff = -1;
+  else
+    int diff = 0;
+  return diff;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
